packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -171,13 +149,22 @@ export const getPolymarketMarkets = async (params: MonitoringParams): Promise<Po
         liquidityNum
         volumeNum
         clobTokenIds
+        endDate
       }
     }
   `;
 
   const { markets: polymarketContracts } = (await request(params.graphqlEndpoint, query)) as any;
 
-  return polymarketContracts.map((contract: { [k: string]: any }) => ({
+  // Remove markets with endDate after 1 week from now.
```

Comment:
We still need to filter the markets, otherwise we have 4.5k which take 30+ mins to populate the [ancillaryData cache](https://github.com/UMAprotocol/protocol/blob/md0x/polymarket-ts-orderbook/packages/monitor-v2/src/monitor-polymarket/common.ts#L178) and allow the bot to run.